### PR TITLE
[WIP] Landing page

### PR DESCRIPTION
**Pull request by princiya**
I need inputs on styling this landing page itself. This PR is an initial draft of what I have in mind and to align with what we want to achieve.

Initial landing page:

Screenshot 2020-07-29 at 00 57 09

Error state:

Screenshot 2020-07-29 at 00 56 24

### CODE REVIEW

File: src/ui/components/App.js [hunk 1]
```diff
@@ -111,22 +132,68 @@ class App extends React.Component {
     return (
       <>
         <div id="header">
-          <div className="logo"></div>
-          <div id="status"></div>
+          <div className="logo" />
+          <div id="status" ref={this.statusRef} />
```
Comment (princiya) [inline src/ui/components/App.js:136]: I tried to have the error state in the landing page itself as an error message above the input element. But the error message comes from `protocol.socket` :

```
function setStatus(text) {
  document.getElementById("status").innerText = text;
}
```

This function uses `innerText`, and I was not able to get the `innerText` value in `componentDidMount` or `componentDidUpdate`. I will try a bit more, else will revert all unnecessary changes like `ref`, `this.state.recordingInputError`

File: src/ui/components/App.js [hunk 2]
```diff
@@ -45,23 +46,43 @@ function setTheme(theme) {
...
+    const params = new URLSearchParams(document.location.search.substring(1));
+    this.setState({
+      recordingId: params.get("id"),
+    });
   }
 
-  componentDidUpdate(prevProps) {
+  componentDidUpdate(prevProps, prevState) {
     const { theme } = this.props;
+    const { recordingId, recordingInputError } = this.state;
```
Comment (princiya) [inline src/ui/components/App.js:70]: Trying to have a better error message, else will remove this unnecessary additional state variable `recordingInputError`.

File: src/ui/components/App.js [hunk 3]
```diff
@@ -111,22 +132,68 @@ class App extends React.Component {
...
+            minSize="20%"
+            maxSize="80%"
+            vert={vert}
+            onResizeEnd={num => {}}
+            startPanel={startPanel}
+            endPanelControl={false}
+            endPanel={endPanel}
+          />
+        ) : (
+          <section className="landing-page-wrapper">
```
Comment (jasonLaster) [inline src/ui/components/App.js:153]: lets move this to its own `renderLocalLandingPage()` function

File: src/ui/components/App.js (hunk 1 above)
Comment (jasonLaster) [reply src/ui/components/App.js:136]: hmm, feels like we should move this to react :)

File: src/ui/components/App.js [hunk 4]
```diff
@@ -45,23 +46,43 @@ function setTheme(theme) {
...
   };
 
   componentDidMount() {
     const { theme } = this.props;
     setTheme(theme);
+
+    const params = new URLSearchParams(document.location.search.substring(1));
+    this.setState({
+      recordingId: params.get("id"),
+    });
```
Comment (jasonLaster) [inline src/ui/components/App.js:65]: I wonder if we need state here or if we can just do something with an uncontrolled input... not sure

File: src/ui/utils/messages.js [hunk 5]
```diff
@@ -0,0 +1,4 @@
+export const messages = {
+  noRecording: "Recording ID not specified",
+  errorRecording: "Error: Invalid recording ID",
```
Comment (jasonLaster) [inline src/ui/utils/messages.js:3]: I think it's okay to have text in the JSX. We're not translating anytime soon

File: src/ui/components/App.js (hunk 1 above)
Comment (princiya) [reply src/ui/components/App.js:136]: yes.................... but it was a nice exercise for me to understand to workflow..

**Comment by princiya** [comment]
> Overall love it!
> 
> One interesting point is that it would be nice if this _only_ happened locally. basically a dev-mode thing. I think we have a helper for this somewhere...

got it. yes the helper function is there in bootstrap.js.

File: src/ui/components/App.js (hunk 4 above)
Comment (princiya) [reply src/ui/components/App.js:65]: yes, this is also part of trying to get that error message within the landing page. i will revert these changes if it is ok to show the error message like it appears on the 2nd attached screenshot.

**Comment by jasonLaster** [comment]
Closing as we now have a reasonable landing page

**Pull request closed without merging**